plugin/contrib/mesos/pkg/executor/executor.go

```diff
@@ -231,6 +248,81 @@ func (k *KubernetesExecutor) onInitialRegistration() {
 	}
 }
 
+// InitializeStaticPodsSource blocks until initial regstration is complete and then creates a static pod source
+// using the given factory func.
+func (k *KubernetesExecutor) InitializeStaticPodsSource(sourceFactory func(string)) {
+	<-k.initialRegComplete
+	staticPodsConfigPath := k.extractStaticPodConfig()
+	if staticPodsConfigPath != "" {
+		log.V(2).Infof("initializing static pods source factory, configured at path %q", staticPodsConfigPath)
+		sourceFactory(staticPodsConfigPath)
+	}
+}
+
+// ExtractStaticPodConfig extracts a compressed archive from k.staticPodsConfig and returns
+// the path to the extracted configuration. any errors are logged and result in returning ""
+func (k *KubernetesExecutor) extractStaticPodConfig() string {
+	if k.staticPodsConfig == nil {
+		return ""
+	}
+
+	// extract to local sendBox so it is automatically cleaned up
+	dir, err := ioutil.TempDir(".", "executor-k8sm-archive")
```

Comment:
In fact I would like it to stay in the sandbox, but rename it to "static-pods". We can pass that path in from the service. In the test cases we pass a temporary directory instead.